etc/build_gstreamer.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# Set error handling
+set -e
+
+# Check if the path argument is provided
+if [ -z "$1" ]; then
+    echo "Usage: $0 <path>"
+    exit 1
+fi
+
+# Change directory to the provided path
+cd "$1" || exit 1
+
+# Clone gstreamer repository
+git clone --depth=1 https://gitlab.freedesktop.org/gstreamer/gstreamer.git
+cd gstreamer || exit 1
+
+# Build gstreamer
+meson setup ../gst-build
```

Comment:
```suggestion
meson setup --wrap-mode=nodownload ../gst-build
```

This will ensure that meson doesn't download any dependencies using wrap files. You should continue to use ubuntu for dependencies.

Also, once you have gotten the build going, if you want to optimize the build time, you can use `-Dauto_features=disabled` and then manually enable the features/plugins you need. Note that this can be tricky.